From 436b252fefd026d58e1161bcb733c8deb26dc8ba Mon Sep 17 00:00:00 2001
From: Debarshi Ray <debarshir@gnome.org>
Date: Wed, 18 Dec 2013 16:39:41 +0100
Subject: [PATCH] libtracker-extract: Link against libicu when using it

---
 src/libtracker-extract/Makefile.am | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/src/libtracker-extract/Makefile.am b/src/libtracker-extract/Makefile.am
index 4c59921..b7f0df9 100644
--- a/src/libtracker-extract/Makefile.am
+++ b/src/libtracker-extract/Makefile.am
@@ -11,6 +11,12 @@ AM_CPPFLAGS =                                          \
 
 lib_LTLIBRARIES = libtracker-extract-@TRACKER_API_VERSION@.la
 
+libtracker_extract_@TRACKER_API_VERSION@_la_LIBADD =   \
+	$(top_builddir)/src/libtracker-sparql-backend/libtracker-sparql-@TRACKER_API_VERSION@.la \
+	$(top_builddir)/src/libtracker-common/libtracker-common.la \
+	$(BUILD_LIBS)                                  \
+	$(LIBTRACKER_EXTRACT_LIBS)
+
 libtracker_extractincludedir=$(includedir)/tracker-$(TRACKER_API_VERSION)/libtracker-extract/
 
 libtracker_extract_@TRACKER_API_VERSION@_la_SOURCES =  \
@@ -63,18 +69,16 @@ if HAVE_LIBICU
 libtracker_extract_@TRACKER_API_VERSION@_la_SOURCES += \
 	tracker-encoding-libicu.c                      \
 	tracker-encoding-libicu.h
+AM_CPPFLAGS +=                                         \
+	$(LIBICU_CFLAGS)
+libtracker_extract_@TRACKER_API_VERSION@_la_LIBADD +=  \
+	$(LIBICU_LIBS)
 endif
 
 libtracker_extract_@TRACKER_API_VERSION@_la_LDFLAGS =  \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-export-symbols-regex '^tracker_.*'
 
-libtracker_extract_@TRACKER_API_VERSION@_la_LIBADD =   \
-	$(top_builddir)/src/libtracker-sparql-backend/libtracker-sparql-@TRACKER_API_VERSION@.la \
-	$(top_builddir)/src/libtracker-common/libtracker-common.la \
-	$(BUILD_LIBS)                                  \
-	$(LIBTRACKER_EXTRACT_LIBS)
-
 # Introspection foo
 -include $(INTROSPECTION_MAKEFILE)
 INTROSPECTION_GIRS =
-- 
1.8.4.2

