--- Makefile.am	2006-11-20 23:57:45.000000000 -0500
+++ Makefile.am.new	2006-11-20 23:58:00.000000000 -0500
@@ -1,7 +1,9 @@
 SUBDIRS = data src filters thumbnailers docs po
 
 pkgconfigdir = $(libdir)/pkgconfig
+xdgconfigdir = $(sysconfdir)/xdg/autostart
 pkgconfig_DATA = tracker.pc
+xdgconfig_DATA = trackerd.desktop
 
 debian_files = \
 	debian/changelog \
@@ -34,12 +36,11 @@
 
 EXTRA_DIST = \
 	$(pkgconfig_DATA) \
+	$(xdgconfig_DATA) \
 	AUTHORS NEWS INSTALL COPYING README \
 	ChangeLog \
 	configure.in \
 	rdf-query-examples/*.rdf \
-	trackerd.desktop \
-	trackerd-kde.desktop \
 	intltool-extract.in \
 	intltool-merge.in \
 	intltool-update.in \
--- trackerd.desktop	2006-11-15 19:08:17.000000000 -0500
+++ trackerd.desktop.new	2006-11-20 20:37:59.000000000 -0500
@@ -8,5 +8,8 @@
 Terminal=false
 Type=Application
 Categories=
-OnlyShowIn=GNOME;
 X-GNOME-Autostart-enabled=true
+X-KDE-autostart-after=panel
+X-KDE-StartupNotify=false
+X-KDE-UniqueApplet=true
+NoDisplay=true
--- src/tracker-search-tool/tracker-search-tool.desktop.in	2006-11-20 08:11:41.000000000 -0500
+++ src/tracker-search-tool/tracker-search-tool.desktop.in.new	2006-11-20 20:24:40.000000000 -0500
@@ -6,6 +6,6 @@
 Icon=gnome-searchtool
 Terminal=false
 Type=Application
-Categories=GTK;Application;Utility;
+Categories=GTK;Utility;
 StartupNotify=true
-NoDisplay=true 
+NoDisplay=true
