--- configure.ac	2011-04-08 08:03:30.000000000 -0400
+++ configure.ac.new	2011-04-09 02:35:13.969905763 -0400
@@ -468,7 +468,7 @@
 TRACKER_NAUTILUS_EXTENSION_REQUIRED="glib-2.0       >= $GLIB_REQUIRED
                                      gio-unix-2.0   >= $GLIB_REQUIRED
                                      gthread-2.0    >= $GLIB_REQUIRED
-                                     gtk+-2.0       >= $GTK_REQUIRED
+                                     gtk+-3.0       >= $GTK_REQUIRED
                                      libnautilus-extension"
 
 PKG_CHECK_MODULES(TRACKER_NAUTILUS_EXTENSION, [$TRACKER_NAUTILUS_EXTENSION_REQUIRED],
