diff -up tracker-0.6.2/src/trackerd/tracker-apps.c.getenv tracker-0.6.2/src/trackerd/tracker-apps.c
--- tracker-0.6.2/src/trackerd/tracker-apps.c.getenv	2007-08-18 20:31:41.000000000 -0400
+++ tracker-0.6.2/src/trackerd/tracker-apps.c	2007-09-11 12:47:38.000000000 -0400
@@ -18,7 +18,8 @@
  * Boston, MA  02110-1301, USA.
  */
 
-#include<string.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include "tracker-apps.h"
 #include "tracker-db.h"
@@ -51,6 +52,7 @@ tracker_applications_add_service_directo
 		value = getenv ("XDG_DATA_DIRS");
 		if (value != NULL) {
 			gchar **dir_array;
+			tracker_log ("XDG_DATA_DIRS is at %p",value);
 			dir_array = g_strsplit (value, ":", 0);
 
 			gint i;
