--- docs/tools/Makefile.am	2010-09-09 06:37:06.000000000 -0400
+++ docs/tools/Makefile.am.new	2010-09-09 10:44:10.022458044 -0400
@@ -23,9 +23,12 @@
 ontology_graph_LDADD =							\
 	$(top_builddir)/src/libtracker-data/libtracker-data.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libtracker-sparql/libtracker-sparql-@TRACKER_API_VERSION@.la \
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 ttl2sgml_SOURCES =							\
 	ttl2sgml.c							\
@@ -41,8 +44,11 @@
 ttl2sgml_LDADD = 							\
 	$(top_builddir)/src/libtracker-data/libtracker-data.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libtracker-sparql/libtracker-sparql-@TRACKER_API_VERSION@.la \
 	$(DBUS_LIBS)							\
 	$(GLIB2_LIBS)							\
-	$(GIO_LIBS)
+	$(GIO_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 EXTRA_DIST = gen-doc.sh
--- docs/tools/Makefile.in	2010-09-09 10:19:42.000000000 -0400
+++ docs/tools/Makefile.in.new	2010-09-09 10:44:10.023458812 -0400
@@ -394,9 +394,12 @@
 ontology_graph_LDADD = \
 	$(top_builddir)/src/libtracker-data/libtracker-data.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libtracker-sparql/libtracker-sparql-@TRACKER_API_VERSION@.la \
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 ttl2sgml_SOURCES = \
 	ttl2sgml.c							\
@@ -412,9 +415,12 @@
 ttl2sgml_LDADD = \
 	$(top_builddir)/src/libtracker-data/libtracker-data.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libtracker-sparql/libtracker-sparql-@TRACKER_API_VERSION@.la \
 	$(DBUS_LIBS)							\
 	$(GLIB2_LIBS)							\
-	$(GIO_LIBS)
+	$(GIO_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-data/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 EXTRA_DIST = gen-doc.sh
 all: all-am
--- docs/reference/libtracker-client/Makefile.am	2010-04-15 06:14:08.000000000 -0400
+++ docs/reference/libtracker-client/Makefile.am.new	2010-04-19 09:56:26.248325370 -0400
@@ -40,7 +40,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-client/Makefile.in	2010-04-15 11:51:12.000000000 -0400
+++ docs/reference/libtracker-client/Makefile.in.new	2010-04-19 09:57:14.958188757 -0400
@@ -364,7 +364,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-client/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-extract/Makefile.am	2010-03-29 10:54:50.000000000 -0400
+++ docs/reference/libtracker-extract/Makefile.am.new	2010-04-19 10:02:46.533447803 -0400
@@ -34,7 +34,9 @@
 GTKDOC_LIBS =								\
 	$(top_builddir)/src/libtracker-extract/libtracker-extract-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-extract/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-extract/Makefile.in	2010-04-15 11:51:12.000000000 -0400
+++ docs/reference/libtracker-extract/Makefile.in.new	2010-04-19 10:03:29.565082879 -0400
@@ -357,7 +357,9 @@
 GTKDOC_LIBS = \
 	$(top_builddir)/src/libtracker-extract/libtracker-extract-@TRACKER_API_VERSION@.la \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-extract/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-miner/Makefile.am	2010-04-15 06:14:08.000000000 -0400
+++ docs/reference/libtracker-miner/Makefile.am.new	2010-04-19 10:04:20.494435581 -0400
@@ -50,7 +50,10 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-miner/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-sparql/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-miner/Makefile.in	2010-04-15 11:51:12.000000000 -0400
+++ docs/reference/libtracker-miner/Makefile.in.new	2010-04-19 10:05:07.469082598 -0400
@@ -374,7 +374,10 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-miner/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-sparql/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-sparql/Makefile.am	2010-09-09 06:37:06.000000000 -0400
+++ docs/reference/libtracker-sparql/Makefile.am.new	2010-09-09 11:55:36.723180352 -0400
@@ -38,7 +38,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-sparql/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
--- docs/reference/libtracker-sparql/Makefile.in	2010-09-09 10:19:42.000000000 -0400
+++ docs/reference/libtracker-sparql/Makefile.in.new	2010-09-09 11:59:35.941180003 -0400
@@ -374,7 +374,9 @@
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(DBUS_LIBS)							\
 	$(GIO_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-sparql/.libs		\
+	-Wl,-rpath,$(top_builddir)/src/libtracker-common/.libs
 
 
 # Extra options to supply to gtkdoc-mkdb
