diff -up tracker-0.8.17/src/tracker-extract/tracker-extract-pdf.c.poppler15 tracker-0.8.17/src/tracker-extract/tracker-extract-pdf.c
--- tracker-0.8.17/src/tracker-extract/tracker-extract-pdf.c.poppler15	2010-10-06 16:23:04.353101001 -0400
+++ tracker-0.8.17/src/tracker-extract/tracker-extract-pdf.c	2010-10-06 16:23:36.238100975 -0400
@@ -243,7 +243,7 @@ extract_content (PopplerDocument *docume
 		rect.x1 = rect.y1 = 0;
 		poppler_page_get_size (page, &rect.x2, &rect.y2);
 
-		text = poppler_page_get_text (page, POPPLER_SELECTION_WORD, &rect);
+		text = poppler_page_get_selected_text (page, POPPLER_SELECTION_WORD, &rect);
 		t = tracker_text_normalize (text, n_words - words, &normalized_words);
 
 		words += normalized_words;
