--- src/tracker-utils/Makefile.am	2010-02-04 05:25:44.000000000 -0500
+++ src/tracker-utils/Makefile.am.new	2010-02-10 12:14:03.649870479 -0500
@@ -15,9 +15,11 @@
 	$(top_builddir)/src/libtracker/libtrackerclient.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(top_builddir)/src/libtracker-db/libtracker-db.la		\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GIO_LIBS)							\
 	$(GCOV_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	$(DBUS_LIBS)
 
 bin_PROGRAMS = 								\
 	tracker-search 							\
--- src/tracker-utils/Makefile.in	2010-02-04 05:56:50.000000000 -0500
+++ src/tracker-utils/Makefile.in.new	2010-02-10 12:13:10.646869409 -0500
@@ -386,9 +386,11 @@
 	$(top_builddir)/src/libtracker/libtrackerclient.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
 	$(top_builddir)/src/libtracker-db/libtracker-db.la		\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GIO_LIBS)							\
 	$(GCOV_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	$(DBUS_LIBS)
 
 tracker_search_SOURCES = tracker-search.c
 tracker_search_LDADD = $(libs)
--- src/tracker-extract/Makefile.am	2010-02-04 05:25:44.000000000 -0500
+++ src/tracker-extract/Makefile.am.new	2010-02-10 12:37:07.051869630 -0500
@@ -161,6 +161,7 @@
 libextract_ps_la_LDFLAGS = $(module_flags)
 libextract_ps_la_LIBADD = 						\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GLIB2_LIBS)                                                   \
 	$(GCOV_LIBS)
 
@@ -242,10 +243,13 @@
 
 tracker_extract_LDADD = 						\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GMODULE_LIBS)							\
 	$(GTHREAD_LIBS)							\
 	$(GCOV_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	$(GIO_LIBS)							\
+	$(DBUS_LIBS)
 
 dbus_sources = 								\
 	tracker-extract-glue.h
--- src/tracker-extract/Makefile.in	2010-02-04 05:56:50.000000000 -0500
+++ src/tracker-extract/Makefile.in.new	2010-02-10 12:36:39.011869844 -0500
@@ -662,6 +662,7 @@
 libextract_ps_la_LDFLAGS = $(module_flags)
 libextract_ps_la_LIBADD = \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GLIB2_LIBS)                                                   \
 	$(GCOV_LIBS)
 
@@ -736,10 +737,13 @@
 
 tracker_extract_LDADD = \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GMODULE_LIBS)							\
 	$(GTHREAD_LIBS)							\
 	$(GCOV_LIBS)							\
-	$(GLIB2_LIBS)
+	$(GLIB2_LIBS)							\
+	$(GIO_LIBS)							\
+	$(DBUS_LIBS)
 
 dbus_sources = \
 	tracker-extract-glue.h
--- src/tracker-applet/Makefile.am	2010-02-04 05:25:44.000000000 -0500
+++ src/tracker-applet/Makefile.am.new	2010-02-10 12:50:09.586869530 -0500
@@ -16,9 +16,11 @@
 tracker_applet_LDADD = 							\
 	$(top_builddir)/src/libtracker/libtrackerclient.la 		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(TRACKER_APPLET_LIBS)						\
 	$(LIBGLADE_LIBS)						\
-	$(GCOV_LIBS)
+	$(GCOV_LIBS)							\
+	-lX11
 
 tracker-marshal.h: tracker-marshal.list
 	$(GLIB_GENMARSHAL) $< --prefix=tracker_marshal --header > $@
--- src/tracker-applet/Makefile.in	2010-02-04 05:56:49.000000000 -0500
+++ src/tracker-applet/Makefile.in.new	2010-02-10 12:49:30.943869990 -0500
@@ -364,9 +364,11 @@
 tracker_applet_LDADD = \
 	$(top_builddir)/src/libtracker/libtrackerclient.la 		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(TRACKER_APPLET_LIBS)						\
 	$(LIBGLADE_LIBS)						\
-	$(GCOV_LIBS)
+	$(GCOV_LIBS)							\
+	-lX11
 
 marshal_sources = \
         tracker-marshal.h                             			\
--- src/tracker-preferences/Makefile.am	2010-02-04 05:25:44.000000000 -0500
+++ src/tracker-preferences/Makefile.am.new	2010-02-10 13:05:09.906902848 -0500
@@ -35,6 +35,7 @@
 tracker_preferences_LDADD = 						\
 	$(top_builddir)/src/libtracker/libtrackerclient.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(LIBGLADE_LIBS)						\
 	$(GTK2_LIBS) 							\
 	$(DBUS_LIBS) 							\
--- src/tracker-preferences/Makefile.in	2010-02-04 05:56:50.000000000 -0500
+++ src/tracker-preferences/Makefile.in.new	2010-02-10 13:05:35.819870342 -0500
@@ -381,6 +381,7 @@
 tracker_preferences_LDADD = \
 	$(top_builddir)/src/libtracker/libtrackerclient.la		\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(LIBGLADE_LIBS)						\
 	$(GTK2_LIBS) 							\
 	$(DBUS_LIBS) 							\
--- utils/albumart/Makefile.am	2010-02-04 05:25:44.000000000 -0500
+++ utils/albumart/Makefile.am.new	2010-02-10 13:25:36.531882295 -0500
@@ -14,5 +14,6 @@
 
 albumart_strip_LDADD =							\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GLIB2_LIBS)
 
--- utils/albumart/Makefile.in	2010-02-04 05:56:51.000000000 -0500
+++ utils/albumart/Makefile.in.new	2010-02-10 13:26:00.227870154 -0500
@@ -333,6 +333,7 @@
 
 albumart_strip_LDADD = \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GLIB2_LIBS)
 
 all: all-am
--- utils/mp3-genre-leading-uppercase/Makefile.am	2010-02-04 05:25:44.000000000 -0500
+++ utils/mp3-genre-leading-uppercase/Makefile.am.new	2010-02-10 13:26:34.142895903 -0500
@@ -14,5 +14,6 @@
 
 mp3_genre_leading_uppercase_LDADD =					\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GLIB2_LIBS)
 
--- utils/mp3-genre-leading-uppercase/Makefile.in	2010-02-04 05:56:51.000000000 -0500
+++ utils/mp3-genre-leading-uppercase/Makefile.in.new	2010-02-10 13:26:57.028869670 -0500
@@ -335,6 +335,7 @@
 
 mp3_genre_leading_uppercase_LDADD = \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(GLIB2_LIBS)
 
 all: all-am
--- utils/qdbm/Makefile.am	2010-02-04 05:25:44.000000000 -0500
+++ utils/qdbm/Makefile.am.new	2010-02-10 13:27:54.475886500 -0500
@@ -15,6 +15,7 @@
 print_LDADD = 								\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
 	$(top_builddir)/src/libtracker-db/libtracker-db.la	 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(QDBM_LIBS) 							\
 	$(GLIB2_LIBS)							
 
@@ -24,6 +25,7 @@
 search_LDADD = 								\
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
 	$(top_builddir)/src/libtracker-db/libtracker-db.la	 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(QDBM_LIBS) 							\
 	$(GLIB2_LIBS)							
 
--- utils/qdbm/Makefile.in	2010-02-04 05:56:52.000000000 -0500
+++ utils/qdbm/Makefile.in.new	2010-02-10 13:28:16.643927434 -0500
@@ -341,6 +341,7 @@
 print_LDADD = \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
 	$(top_builddir)/src/libtracker-db/libtracker-db.la	 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(QDBM_LIBS) 							\
 	$(GLIB2_LIBS)							
 
@@ -350,6 +351,7 @@
 search_LDADD = \
 	$(top_builddir)/src/libtracker-common/libtracker-common.la 	\
 	$(top_builddir)/src/libtracker-db/libtracker-db.la	 	\
+	$(top_builddir)/src/libstemmer/libstemmer.la	 		\
 	$(QDBM_LIBS) 							\
 	$(GLIB2_LIBS)							
 
